src/test/java/org/opensearch/knn/KNNSingleNodeTestCase.java

```diff
@@ -181,6 +181,38 @@ protected void addDoc(String index, String docId, String fieldName, String dummy
         assertEquals(response.status(), RestStatus.CREATED);
     }
 
+    /**
+     * Index a new model
+     */
+    protected void addDoc(Model model) throws IOException, ExecutionException, InterruptedException {
```

Comment:
Can we actually create a model instead of mimicking it by adding model metadata in system index?
Also, this method should not be named as `addDoc`. Should it be like `writeModelToModelSystemIndex`?